src/containers/DataDisplay/DataViewTable.tsx

```diff
@@ -108,7 +120,8 @@ const DataViewTableContent: React.FC<DataViewTableContentProps> = ({
   const showToast = useToast({ style: 'negative' })
 
   const { headerRow, tableRows, totalCount } = dataViewTable
-  const showDetailsForRow = (id: number) => push(`/data/${dataViewCode}/${id}`)
+  const showDetailsForRow = (id: number) =>
+    push(`/data/${dataViewCode}/${id}`, { dataTableFilterQuery: location.search })
```

Comment:
This passes the current filter (url query) state to the detail page, so when we click the "back to Table" link, it restores the correct filter state.